### DISCUSSION TITLE
How to interpret a model?

### DISCUSSION THREAD

**Discussion by Aaryanverma** [online-ml/river#648] [Q&A]
For any model like logistic regression or any tree-based model, how can we interpret a model just like SHAP or LIME can be used in scikit-learn models? As far as I know, SHAP/LIME cannot be used at the moment because of incompatibility with river models.

**Comment by Noazao** [comment]
@Aaryanverma By any chance, did you manage to implement SHAP with river ? I am try to interpret a HalfSpaceTree model. I shows that river `[Model type not yet supported by TreeExplainer: <class 'river.anomaly.hst.HalfSpaceTrees'>]()`

**Reply by Aaryanverma** [reply]
@Noazao I managed to implement LIME with river by creating a wrapper class around river model to make it look like a sklearn model.

Refer here: https://marcotcr.github.io/lime/tutorials/Tutorial_H2O_continuous_and_cat.html#Wrapper-class

**Reply by Noazao** [reply]
@Aaryanverma Great! thank you.
